src/Client/Repository.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Tuf\Client;
+
+use Tuf\Exception\RepoFileNotFound;
+use Tuf\Loader\SizeCheckingLoader;
+use Tuf\Metadata\RootMetadata;
+use Tuf\Metadata\SnapshotMetadata;
+use Tuf\Metadata\TargetsMetadata;
+use Tuf\Metadata\TimestampMetadata;
+
+/**
+ * Defines a backend to load untrusted TUF metadata objects.
+ */
+class Repository
+{
+    /**
+     * The maximum number of bytes to download if the remote file size is not
+     * known.
+     *
+     * @var int
+     */
+    public const MAX_BYTES = 100000;
+
+    public function __construct(private SizeCheckingLoader $loader)
+    {
+    }
+
+    /**
+     * Loads untrusted root metadata.
+     *
+     * @param int $version
+     *   The version of the root metadata to load.
+     *
+     * @return \Tuf\Metadata\RootMetadata|null
+     *   An instance of \Tuf\Metadata\RootMetadata, or null if the requested
+     *   version of the metadata doesn't exist.
+     */
+    public function getRoot(int $version): ?RootMetadata
+    {
+        try {
+            $data = $this->loader->load("$version.root.json", self::MAX_BYTES);
```

Comment:
It seems like if the class property wasn't named `loader` generically it would be clear what is going on here. could

Reply:
How 'bout `dataLoader` as a name for the class property?

Reply:
Renamed to sizeCheckingLoader per discussion on Zoom.